packages/components/src/range-control/index.js

```diff
@@ -39,263 +35,257 @@ import {
-
-		const isValueReset = value === null;
-		const currentValue = value !== undefined ? value : currentInput;
-
-		const inputSliderValue = isValueReset ? '' : currentValue;
-
-		const rangeFillValue = isValueReset
-			? floatClamp( max / 2, min, max )
-			: value;
-
-		const calculatedFillValue = ( ( value - min ) / ( max - min ) ) * 100;
-		const fillValue = isValueReset ? 50 : calculatedFillValue;
-		const fillValueOffset = `${ clamp( fillValue, 0, 100 ) }%`;
-
-		const classes = classnames( 'components-range-control', className );
-
-		const wrapperClasses = classnames(
-			'components-range-control__wrapper',
-			!! marks && 'is-marked'
-		);
-
-		const id = `inspector-range-control-${ instanceId }`;
-
-		const describedBy = !! help ? `${ id }__help` : undefined;
-		const enableTooltip = showTooltipProp !== false && isFinite( value );
-
-		const handleOnChange = ( event ) => {
-			const nextValue = parseFloat( event.target.value );
-
-			if ( isNaN( nextValue ) ) {
-				handleOnReset();
-				return;
-			}
-
-			setValue( nextValue );
-			onChange( nextValue );
-		};
-
-		const handleOnReset = () => {
-			let resetValue = parseFloat( resetFallbackValue );
-			let onChangeResetValue = resetValue;
-
-			if ( isNaN( resetValue ) ) {
-				resetValue = null;
-				onChangeResetValue = undefined;
-			}
-
-			setValue( resetValue );
-
-			/**
-			 * Previously, this callback would always receive undefined as
-			 * an argument. This behavior is unexpected, specifically
-			 * when resetFallbackValue is defined.
-			 *
-			 * The value of undefined is not ideal. Passing it through
-			 * to internal <input /> elements would change it from a
-			 * controlled component to an uncontrolled component.
-			 *
-			 * For now, to minimize unexpected regressions, we're going to
-			 * preserve the undefined callback argument, except when a
-			 * resetFallbackValue is defined.
-			 */
-			onChange( onChangeResetValue );
-		};
-
-		const handleShowTooltip = () => setShowTooltip( true );
-		const handleHideTooltip = () => setShowTooltip( false );
-
-		const handleOnBlur = ( event ) => {
-			onBlur( event );
-			setIsFocused( false );
-			handleHideTooltip();
-		};
-
-		const handleOnFocus = ( event ) => {
-			onFocus( event );
-			setIsFocused( true );
-			handleShowTooltip();
-		};
-
-		const hoverInteractions = useDebouncedHoverInteraction( {
-			onShow: handleShowTooltip,
-			onHide: handleHideTooltip,
-			onMouseMove,
-			onMouseLeave,
-		} );
-
-		const offsetStyle = {
-			[ isRTL ? 'right' : 'left' ]: fillValueOffset,
-		};
-
-		return (
-			<BaseControl
-				className={ classes }
-				label={ label }
-				id={ id }
-				help={ help }
-			>
-				<Root
-					className="components-range-control__root"
-					isRTL={ isRTL }
+function RangeControl(
+	{
+		afterIcon,
+		allowReset = false,
+		beforeIcon,
+		className,
+		currentInput,
+		color: colorProp = color( 'ui.brand' ),
+		disabled = false,
+		help,
+		initialPosition,
+		isShiftStepEnabled = true,
```

Comment:
I've seen a few changes in props here, so we should make sure docs are in sync. At least, the `isShiftEnabled` prop needs to be added.